src/main/java/seedu/address/logic/commands/CommandResult.java

```diff
@@ -20,18 +23,27 @@
     /**
      * Constructs a {@code CommandResult} with the specified fields.
      */
-    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit) {
+    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit, boolean isClass) {
         this.feedbackToUser = requireNonNull(feedbackToUser);
         this.showHelp = showHelp;
         this.exit = exit;
+        this.isClass = isClass;
     }
 
     /**
      * Constructs a {@code CommandResult} with the specified {@code feedbackToUser},
      * and other fields set to their default value.
      */
     public CommandResult(String feedbackToUser) {
-        this(feedbackToUser, false, false);
+        this(feedbackToUser, false, false, false);
+    }
+
+    /**
+     * Constructs a {@code CommandResult} with the specified {@code feedbackToUser},
+     * and {@code isClass} and other fields set to their default value.
+     */
+    public CommandResult(String feedbackToUser, boolean isClass) {
```

Comment:
would it be okay to remove this, as it might cause confusion with the other boolean fields / can we change it to a factory method?

Reply:
Agreed, thank you!